### PR TITLE
Yards create

### PR DESCRIPTION
**Pull request by Gvieve**
### Add
* new methods for stubbing users on spec tests
* Update Engine create_yard method to parse response
* Update redirect on create to yard show page rather than host dash

### Testing
* User model tests for ominauth
* Feature test for yard create, replace webmocks with VCR on new_spec

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -62,7 +62,8 @@ group :development, :test do
 end
 
 group :test do
-    gem 'webmock'
+  gem 'webmock'
+  gem 'vcr'
 end
```
Comment (amsmyth1) [inline Gemfile:67]: VCR! VCR! VCR!

File: spec/features/host/dashboard/index_spec.rb [hunk 2]
```diff
@@ -2,66 +2,53 @@
...
-        to_return(status: 200, body: response, headers: {})
-
-    visit root_path
-    click_button 'Login through Google'
-    @user = User.find_by(uid: 123545)
+    user = User.create!(id:1, uid: '123545', username: 'Dominic Padula', email:'thisemail@gmail.com', password: SecureRandom.hex(15) )
+    omniauth_response = stub_omniauth_happy('123545', 'Dominic Padula', 'thisemail@gmail.com')
+    @user_1 = User.from_omniauth(omniauth_response)
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user_1)
   end
```
Comment (amsmyth1) [inline spec/features/host/dashboard/index_spec.rb:9]: I would love if you could run through how this is working. Thanks for figuring this out completely! Testing and user paths were going to plague the frontend.

File: spec/features/host/yards/new_spec.rb [hunk 3]
```diff
@@ -2,100 +2,79 @@
...
-          headers: {
-         'Accept'=>'*/*',
-         'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
-         'Content-Type'=>'application/x-www-form-urlencoded',
-         'User-Agent'=>'Faraday v1.3.0'
-          }).
-        to_return(status: 200, body: "", headers: {})
-
-        visit root_path
-        click_button 'Login through Google'
```
Comment (amsmyth1) [inline spec/features/host/yards/new_spec.rb:39]: Seeing this red makes me so happy.

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -42,19 +42,13 @@
   # instead of true.
   config.use_transactional_fixtures = true
   OmniAuth.config.test_mode = true
-  # OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new({
```
Comment (amsmyth1) [inline spec/rails_helper.rb:45]: Now we pass parameters instead of building into our helper method? nice!

**Review by amsmyth1 — APPROVED** [review]
This is awesome. Great work figuring out how to get testing and omniauth working together properly. It took a lot of days and brainpower to get here but this is the foundation frontend application needed. Great work team!

File: spec/features/host/dashboard/index_spec.rb (hunk 2 above)
Comment (domo2192) [reply spec/features/host/dashboard/index_spec.rb:9]: Nice fix

File: app/services/engine_service.rb [hunk 5]
```diff
@@ -9,8 +9,10 @@ def self.all_purposes
   def self.create_yard(yard_params)
     response = connection.post('/api/v1/yards') do |req|
       req.headers["CONTENT_TYPE"] = "application/json"
-      req.body = JSON.generate(yard: yard_params)
+      req.params = yard_params
     end
```
Comment (domo2192) [inline app/services/engine_service.rb:13]: Hopefully we can go into this soon why this solution works

**Review by domo2192 — APPROVED** [review]

**Pull request merged**